### ISSUE TITLE
Debug magic fails to parse breakpoint files with colon in path

### ISSUE THREAD

**Issue by jlumpe** [ipython/ipython#9932]
The `%debug` magic has a `breakpoint` option formatted like `--breakpoint FILE:LINE`, where the file name and line number are separated by a colon. This command fails when a windows-style path containing a colon is used, e.g. beginning with `c:\`.

Example:

```
>>> mymodule.__file__
'c:\\users\\me\\git\\project\\mymodule.py'
>>> %debug --breakpoint {mymodule.__file__}:123 mymodule.foo()
ValueError...
```

The error occurs [here:](https://github.com/ipython/ipython/blob/5c9c918bc0a11057184ec143da13b68994f59666/IPython/core/magics/execution.py#L441)

```
    if breakpoint:
        (filename, bp_line) = breakpoint.split(':', 1)
        bp_line = int(bp_line)
```

where the breakpoint argument is incorrectly split at the first colon resulting in `bp_line` containing part of the file path. I think changing line 440 to

```
 (filename, bp_line) = breakpoint.rsplit(':', 1)
```

to split at the last colon should work just fine.

**Comment by Carreau** [comment]
Thanks ! We'll have a look. 

@thesageinpilani you wanted something a bit more complicated; do you want to have a look at this one ?

It should not be that hard to fix, but figuring out where to do it might scratch your head a bit. 
I don't give you any hints on purpose, but feel free to ask here when you have any questions; I will be happy to describe things in more details.

**Comment by thesageinpilani** [comment]
sure sir, i will try my best.

**Comment by Carreau** [comment]
@thesageinpilani you can use the following to try:

```
In [1]: with open('mymodule.py', 'w') as f:
   ...:     f.write("""
   ...: def foo():
   ...:     1+1
   ...:     """)

In [2]: import mymodule

In [3]: %debug --breakpoint {mymodule.__file__}:3 mymodule.foo()
```

@jlumpe gave also a lot of indication on where and how to fix that. 

If you have any questions on how this things works, feel free to ask.

**Comment by Carreau** [comment]
Closed by #9953 and backported.
